numpy/distutils/fcompiler/environment.py

```diff
@@ -49,10 +49,24 @@ def get(self, name, default=None):
         return var
 
     def _get_var(self, name, conf_desc):
-        hook, envvar, confvar, convert = conf_desc
+        hook, envvar, confvar, convert, append = conf_desc
         var = self._hook_handler(name, hook)
         if envvar is not None:
-            var = os.environ.get(envvar, var)
+            envvar_contents = os.environ.get(envvar)
+            if envvar_contents is not None:
+                if convert is not None:
+                    envvar_contents = convert(envvar_contents)
+                else:
+                    envvar_contents = [envvar_contents]
+                if append and os.environ.get('NPY_DISTUTILS_APPEND_FLAGS', '0') == '1':
+                    if var is None:
+                        var = ''
```

Comment:
Shouldn't this be `var = []`, in order to avoid an extra space appearing? Does this also remove the need for the try/catch?